pkg/git/providers/git/git.go

```diff
@@ -0,0 +1,69 @@
+package git
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const (
+	GitPasswordTokenEnv       = "GIT_PASSWORD"
+	EksaGitPasswordTokenEnv   = "EKSA_GIT_PASSWORD"
+	EksaGitPrivateKeyTokenEnv = "EKSA_GIT_PRIVATE_KEY"
+)
+
+type Options struct {
+	RepositoryUrl string
+	User          string
+}
+
+func GetGitPasswordFromEnv() (string, error) {
+	password, err := GetTokenFromEnv(EksaGitPasswordTokenEnv)
+	if err != nil {
+		return "", err
+	}
+	if len(password) > 0 {
+		if err := os.Setenv(GitPasswordTokenEnv, password); err != nil {
```

Comment:
Curious, why is this needed? Can't it be passed to the executable as an argument?

Reply:
It can be passed as an argument as well as an env var. we can decide to always send it in as an argument.